### PR TITLE
Addition of Orca `dtype`

### PR DESCRIPTION
**Pull request by muammar**
## Description
Orca `dtype` addition needed along with  https://github.com/MolSSI/QCEngine/pull/178

## Changelog description
- Addition of Orca `dtype. 

## Status

- [x] Code base linted
- [x] Ready to go

### CODE REVIEW

File: qcelemental/molparse/to_string.py [hunk 1]
```diff
@@ -73,6 +74,7 @@ def to_string(
     default_units = {
         "xyz": "Angstrom",
         "xyz+": "Angstrom",
+        "orca": "Angstrom",
```
Comment (loriab) [inline qcelemental/molparse/to_string.py]: want Bohr unless format definition utterly unable to accept it

File: qcelemental/molparse/to_string.py [hunk 2]
```diff
@@ -136,6 +138,24 @@ def to_dict(self) -> Dict:
...
+        atoms = _atoms_formatter(molrec, geom, atom_format, ghost_format, width, prec, 2)
+        nat = len(atoms)
+
+        first_line = """"""
+        smol = [first_line.rstrip()]
+        smol.append(f"{'*xyz '} {int(molrec['molecular_charge'])} {molrec['molecular_multiplicity']}")
+
+        smol.extend(atoms)
+        smol.append("*")
+
```
Comment (loriab) [inline qcelemental/molparse/to_string.py]: good to get some tests here (https://github.com/MolSSI/QCElemental/blob/master/qcelemental/tests/test_molparse_to_string.py#L188)

File: qcelemental/molparse/to_string.py (hunk 1 above)
Comment (muammar) [reply qcelemental/molparse/to_string.py]: Done :)

File: qcelemental/tests/test_molparse_to_string.py [hunk 3]
```diff
@@ -143,6 +143,21 @@
...
+H                     2.000000000000     0.000000000000     0.000000000000
+H                    -2.000000000000     0.000000000000     0.000000000000
+*
+""",
+    "ans2_orca_ang": """
+*xyz 0 2
+Co                    0.000000000000     0.000000000000     0.000000000000
+H                     2.000000000000     0.000000000000     0.000000000000
+H                    -2.000000000000     0.000000000000     0.000000000000
+*
```
Comment (dgasmith) [inline qcelemental/tests/test_molparse_to_string.py]: These two look identical at 2 units?

Comment (muammar) [reply qcelemental/tests/test_molparse_to_string.py]: My bad.

Comment (muammar) [reply qcelemental/tests/test_molparse_to_string.py]: I think this should be solved now.

File: qcelemental/molparse/to_string.py [hunk 4]
```diff
@@ -136,6 +138,26 @@ def to_dict(self) -> Dict:
 
         smol.extend(atoms)
 
+    if dtype == "orca":
```
Comment (loriab) [inline qcelemental/molparse/to_string.py]: `elif` here will fix some tests.

note that if you `conda install pytest`, you can `pytest qcelemental/ -rws -v` in the main clone directory to test locally.

not mandatory, but it'd be nice to test the ghost atoms, too. there's a test you can mirror.

File: qcelemental/molparse/to_string.py [hunk 5]
```diff
@@ -136,6 +138,26 @@ def to_dict(self) -> Dict:
...
+        # Notes
+        # * if units not in umap (e.g., nm), can't be read back in by from_string()
+
+        atom_format = "{elem}" if atom_format is None else atom_format
+        ghost_format = "@{elem}" if ghost_format is None else ghost_format
+        umap = {"bohr": "! Bohrs", "angstrom": None}
```
Comment (loriab) [inline qcelemental/molparse/to_string.py]: I suspect this `None` instead of `""` is going to cause trouble, too.

Comment (muammar) [reply qcelemental/molparse/to_string.py]: Fixed this.

**Comment by loriab** [comment]
Hi @muammar, I edited your branch a bit to get the orca dtype into the qcel release this week. Note that to resolve conflicts I had to force-push it, so don't try to update your local branch with `pull`. Let me know if you disagree with any of my changes.

**Comment by muammar** [comment]
> Hi @muammar, I edited your branch a bit to get the orca dtype into the qcel release this week. Note that to resolve conflicts I had to force-push it, so don't try to update your local branch with `pull`. Let me know if you disagree with any of my changes.

Thanks for merging this into `master` @loriab !

**Pull request merged**